server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -470,6 +480,38 @@ protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener
             .execute(wrappedListener.map(r -> null));
     }
 
+    private void removeExistingIndexBlocks(ActionListener<Void> listener) {
```

Comment:
nit: I don't think we need a listener to encapsulate the call to `checkFinished` - may as well just call `checkFinished` directly instead.

Reply:
True! Done here https://github.com/elastic/elasticsearch/pull/87841/commits/c57209ff949e72989586cac74743576ee542f76e.